src/Compilers/CSharp/Test/Semantic/Semantics/ExtensionTypeTests.cs

```diff
@@ -0,0 +1,4993 @@
+            // (5,33): error CS0216: The operator 'R.operator ==(R, R)' requires a matching operator '!=' to also be defined
+            //     public static bool operator ==(R r1, R r2) => throw null;
+            Diagnostic(ErrorCode.ERR_OperatorNeedsMatch, "==").WithArguments("R.operator ==(R, R)", "!=").WithLocation(5, 33)
+            );
+        // PROTOTYPE need to finalize rules on operators. Equals and GetHashCode can't be overridden...
+    }
+
+    [Theory]
+    [InlineData("class")]
+    [InlineData("struct")]
+    [InlineData("interface")]
+    [InlineData("enum")]
+    public void Members_NoDefaultCtor(string type)
+    {
+        var src = $$"""
+{{type}} UnderlyingType { }
+explicit extension R for UnderlyingType { }
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyDiagnostics();
+
+        var r = comp.GlobalNamespace.GetTypeMember("R");
+        Assert.Empty(r.GetMembers());
+        // PROTOTYPE verify PE symbol too
+    }
+
+    [Fact]
+    public void Members_Operators_UsingUnderlyingType()
+    {
+        // PROTOTYPE need to finalize the rules for operators
+        var src = """
+class UnderlyingClass { }
+explicit extension R for UnderlyingClass
+{
+    public static int operator+(UnderlyingClass c1, UnderlyingClass c2) => throw null;
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyDiagnostics(
+            // (4,31): error CS0563: One of the parameters of a binary operator must be the containing type
+            //     public static int operator+(UnderlyingClass c1, UnderlyingClass c2) => throw null;
+            Diagnostic(ErrorCode.ERR_BadBinaryOperatorSignature, "+").WithLocation(4, 31)
+            );
+
+        var r = comp.GlobalNamespace.GetTypeMember("R");
+        AssertEx.Equal(new[]
+            {
+                "System.Int32 R.op_Addition(UnderlyingClass c1, UnderlyingClass c2)"
+            },
+            r.GetMembers().ToTestDisplayStrings());
+    }
+
+    [Fact]
+    public void Members_StaticExtension()
+    {
+        var src = """
+interface I { }
+class UnderlyingClass : I { }
+static explicit extension R for UnderlyingClass
+{
+    public static int StaticField = 0;
+    const string Const = "hello";
+
+    void Method() { } // 1
+    static void StaticMethod() { }
+
+    int Property { get => throw null; set => throw null; } // 2
+    static int StaticProperty { get => throw null; set => throw null; }
+    int this[int i] => throw null; // 3
+
+    class NestedType { }
+    static class StaticNestedType { }
+    explicit extension NestedR for UnderlyingClass { }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyDiagnostics(
+            // (8,10): error CS0708: 'Method': cannot declare instance members in a static type
+            //     void Method() { } // 1
+            Diagnostic(ErrorCode.ERR_InstanceMemberInStaticClass, "Method").WithArguments("Method").WithLocation(8, 10),
```

Comment:
The message of the diagnostic is fine (it says "static type"). If you meant update the name of the error code, I prefer to leave it as-is to avoid churn to existing tests.

Reply:
This is one of the comments that didn't end up in the right spot. Look at line 499.